### DISCUSSION TITLE
Managing large file editing with diffs using LocalCommandLineCodeExector

### DISCUSSION THREAD

**Discussion by aakashd** [microsoft/autogen#3349] [Ideas]
We have been using LLMs to deal with legacy code which at times have files thousands of lines long. We have found diffs to be the best useful way to get better outcome than asking the LLM to rewrite the whole file. 

We are not using Autogen at the moment, but as I evaluate it, I realize that the LocalCommandLineCodeExecutor does not support diffs. Are there any thoughts about why this capability is not implemented yet OR it's just hasn't been prioritized?

**Comment by Rblasik** [comment]
Whats diffs, may I ask?

**Comment by junkgear** [comment]
https://microsoft.github.io/autogen/dev//reference/python/autogen_ext.memory.canvas.html
